app/controllers/CredentialsAuthController.scala

```diff
@@ -0,0 +1,97 @@
+
+/**
+  * The credentials auth controller.
+  *
+  * @param messagesApi         The Play messages API.
+  * @param env                 The Silhouette environment.
+  * @param userService         The user service implementation.
+  * @param authInfoRepository  The auth info repository implementation.
+  * @param credentialsProvider The credentials provider.
+  * @param configuration       The Play configuration.
+  * @param clock               The clock instance.
+  */
+class CredentialsAuthController @Inject()(
+                                           val messagesApi: MessagesApi,
+                                           val env: Environment[User, JWTAuthenticator],
+                                           userService: UserService,
+                                           authInfoRepository: AuthInfoRepository,
+                                           credentialsProvider: CredentialsProvider,
+                                           configuration: Configuration,
+                                           clock: Clock)
+  extends Silhouette[User, JWTAuthenticator] {
+
+  /**
+    * Converts the JSON into a `SignInForm.Data` object.
+    */
+  implicit val dataReads = (
+    (__ \ 'email).read[String] and
+      (__ \ 'password).read[String] and
+      (__ \ 'rememberMe).read[Boolean]
+    ) (SignInForm.Data.apply _)
+
+  /**
+    * Authenticates a user against the credentials provider.
+    *
+    * @return The result to display.
+    */
+  def authenticate: Action[JsValue] = Action.async(parse.json) { implicit request =>
+    request.body.validate[SignInForm.Data].map { data =>
+
+      logger.debug(s"Authentification Request: $data")
+
+      credentialsProvider.authenticate(Credentials(data.email, data.password)).flatMap { loginInfo =>
+        userService.retrieve(loginInfo).flatMap {
+          case Some(user) => env.authenticatorService.create(loginInfo).map {
+            case authenticator if data.rememberMe =>
+              val c = configuration.underlying
+              authenticator.copy(
+                expirationDateTime = clock.now + c.as[FiniteDuration]("silhouette.authenticator.rememberMe.authenticatorExpiry"),
+                idleTimeout = c.getAs[FiniteDuration]("silhouette.authenticator.rememberMe.authenticatorIdleTimeout")
+              )
+            case authenticator => authenticator
+          }.flatMap { authenticator =>
+            env.eventBus.publish(LoginEvent(user, request, request2Messages))
+            env.authenticatorService.init(authenticator).map { token =>
+              Ok(Json.obj("token" -> token))
+            }
+          }
+          case None => Future.failed(new IdentityNotFoundException("Couldn't find user"))
+        }
+      }.recover {
+        case e: ProviderException => {
+          logger.error(s"ProviderException: $e.getMessage", e)
+          Unauthorized(Json.obj("message" -> Messages("invalid.credentials")))
+        }
+      }
+    }.recoverTotal {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382300/issues/source?bid=3144114&fileBranchId=3144125#l91)